### ISSUE TITLE
unknown and strict-type-predicate

### ISSUE THREAD

**Issue by ThomasdenH** [palantir/tslint#4107]
### Bug Report

- __TSLint version__: 5.11.0
- __TypeScript version__: 3.0.1
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
const body: unknown = 'test';
if (typeof body === 'object') // Expression is always true.
  console.log('a');
```

with `tslint.yaml` configuration:

```yaml
---
rules:
  strict-type-predicates: true
```

#### Actual behavior
tslint falsely warns that the expression is always true.

#### Expected behavior
Since the object is of type `unknown` (and actually a `string`), the expression could be both true and false and thus tslint should not warn.

**Comment by InExtremaRes** [comment]
Same here. In fact `unknown` should be treated the same as `any` with regards to this rule.
